Improve MPI Detection in Conda Environments to Avoid Immediate Fatal Errors

Currently, when building in a Conda environment, if the output of conda list | grep mpi returns no MPI-related information, the build fails immediately with a fatal error. For example, the error output is:

```sh
(base) ┌──(athena㉿DESKTOP-D0DH9M9)-[~/Document/gitkhyops/khiops]
└─$ cd ~/Document/gitkhyops/khiops-build

(base) ┌──(athena㉿DESKTOP-D0DH9M9)-[~/Document/gitkhyops/khiops-build]
└─$ cmake -DCMAKE_BUILD_TYPE=Debug \
      -DMPI_IMPL=OpenMPI \
      -DMPI_EXECUTABLE=/usr/bin/mpiexec \
      -DMPI_CXX_INCLUDE_PATH=/usr/lib/x86_64-linux-gnu/openmpi/include \
      -DMPI_CXX_LIBRARIES=/usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so \
      ../khiops
-- Building Khiops Version: 10.3.0 (CMake version : 10.3.0)
-- The CXX compiler identification is GNU 14.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done

Khiops build options:

  CMAKE_BUILD_TYPE="Debug"
  MPI="ON"
  TESTING="OFF"
  C11="ON"
  BUILD_LEX_YACC="OFF"
  BUILD_JARS="OFF"


-- Auto-detected conda environment
CMake Warning at CMakeLists.txt:84 (message):
  You are building in a conda environment without using `conda build`.
  find_mpi may not work as expected.


-- Executables will be stored in /home/athena/Document/gitkhyops/khiops-build/bin/
-- Libraries will be stored in /home/athena/Document/gitkhyops/khiops-build/lib/
-- Found MPI_CXX: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so (found suitable version "3.1", minimum required is "2.0")
-- Found MPI: TRUE (found suitable version "3.1", minimum required is "2.0")
-- MPI command line: /usr/bin/mpiexec -n 4  EXECUTABLE  ARGS
CMake Error at scripts/get_mpi_implementation.cmake:31 (message):
  Missing information to discover the MPI implementation
Call Stack (most recent call first):
  CMakeLists.txt:150 (get_mpi_implementation)


-- Configuring incomplete, errors occurred!
```

This behavior forces an immediate termination of the configuration even though a system MPI (e.g., OpenMPI) is installed. Instead of terminating, we should fall back to using the provided MPI implementation (via the MPI_IMPL variable) if available.

### Proposed Solution:

Modify the get_mpi_implementation function in scripts/get_mpi_implementation.cmake so that if no MPI library information is detected (i.e., VAR_MPI_INFO is empty or undefined), the script checks whether the user has provided an MPI implementation using MPI_IMPL. If MPI_IMPL is defined, it should use that value instead of generating a fatal error.

Below is the modified code snippet:

```cmake
  # If no MPI library information is detected, then use the provided MPI_IMPL (if set),
  # otherwise, terminate with a fatal error.
  if(NOT DEFINED VAR_MPI_INFO OR "${VAR_MPI_INFO}" STREQUAL "")
    if(DEFINED MPI_IMPL)
      message(STATUS "Using provided MPI_IMPL: ${MPI_IMPL}")
    else()
      message(FATAL_ERROR "Missing information to discover the MPI implementation")
    endif()
  endif()
```
This change allows the build to use the system MPI configuration when Conda does not list any MPI libraries, improving robustness for users building within a Conda environment.

Additional Context:

    Khiops Version: 10.3.0 
    Environment: Building in a Conda environment (without using conda build)
    System MPI: OpenMPI installed on the system

A merge request will be generated soon.

--- a/.github/workflows/run-standard-tests.yml
+++ b/.github/workflows/run-standard-tests.yml
@@ -42,7 +42,7 @@ jobs:
     strategy:
       matrix:
         build-setup:
-          - {os: windows-2022, cmake-preset: windows-msvc}
+          - {os: windows-2025-vs2026, cmake-preset: windows-msvc}
           - {os: ubuntu-latest, cmake-preset: linux-gcc}
           - {os: ubuntu-24.04-arm, cmake-preset: linux-gcc}
           - {os: macos-14, cmake-preset: macos-clang}
@@ -92,7 +92,7 @@ jobs:
       fail-fast: false
       matrix:
         build-setup:
-          - {os: windows-2022, cmake-preset: windows-msvc}
+          - {os: windows-2025-vs2026, cmake-preset: windows-msvc}
           - {os: ubuntu-latest, cmake-preset: linux-gcc}
           - {os: ubuntu-24.04-arm, cmake-preset: linux-gcc}
           - {os: macos-14, cmake-preset: macos-clang}

--- a/.github/workflows/run-unit-tests.yml
+++ b/.github/workflows/run-unit-tests.yml
@@ -28,7 +28,7 @@ jobs:
       fail-fast: false
       matrix:
         build-setup:
-          - {os: windows-2022, cmake-preset: windows-msvc-debug}
+          - {os: windows-2025-vs2026, cmake-preset: windows-msvc-debug}
           - {os: ubuntu-latest, cmake-preset: linux-gcc-debug}
           - {os: macos-latest, cmake-preset: macos-clang-debug}
     runs-on: ${{ matrix.build-setup.os }}

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,13 +111,6 @@ if(DEFINED ENV{CONDA_PREFIX} AND NOT "$ENV{CONDA_PREFIX}" STREQUAL "")
   message(STATUS "Auto-detected conda environment")
 endif()
 
-if(IS_CONDA AND MPI)
-  if(NOT DEFINED ENV{mpi})
-    message(
-      WARNING "You are building in a conda environment without using `conda build`. find_mpi may not work as expected.")
-  endif()
-endif()
-
 if(IS_PIP AND IS_CONDA)
   message(FATAL_ERROR "Building in a conda environment and using scikit-build-core")
 endif()

--- a/scripts/get_mpi_implementation.cmake
+++ b/scripts/get_mpi_implementation.cmake
@@ -18,6 +18,12 @@ function(get_mpi_implementation)
         COMMAND grep mpi
         OUTPUT_VARIABLE VAR_MPI_INFO)
     endif()
+    if(VAR_MPI_INFO STREQUAL "")
+      message(
+        FATAL_ERROR
+          "No MPI implementation found in conda environment. Please make sure to install an MPI implementation (e.g. `conda install openmpi`)"
+      )
+    endif()
   elseif(IS_PIP)
     set(DETECTION_MESSAGE "from pip environment")
 
@@ -39,10 +45,12 @@ function(get_mpi_implementation)
     set(VAR_MPI_INFO "${MPI_LIBRARIES}")
   endif(IS_CONDA)
 
-  # ERROR if VAR_MPI_INFO is not defined, it means either: - in standard environment find_mpi provides no MPI path (MPI
-  # is not installed) - or in conda build, the 'mpi' variable is missing and find_mpi may find the system wide mpi, this
-  # is not not what we want. - or in conda, outside of the build process, the mpi package is not installed and find_mpi
-  # may find the system wide mpi.
+  # cmake-format: off
+  # ERROR if VAR_MPI_INFO is not defined, it means either: 
+  # - in standard environment find_mpi provides no MPI path (MPI is not installed) 
+  # - or in conda build, the 'mpi' variable is missing and find_mpi may find the system wide mpi, this is not not what we want. 
+  # - or in conda, outside of the build process, the mpi package is not installed and find_mpi may find the system wide mpi.
+  # cmake-format: on
   if(NOT DEFINED VAR_MPI_INFO OR "${VAR_MPI_INFO}" STREQUAL "")
     message(FATAL_ERROR "Missing information to discover the MPI implementation")
   endif()

